sycl/include/CL/sycl/detail/pi.hpp

```diff
@@ -121,72 +119,88 @@ void printArgs(Arg0 arg0, Args... args) {
 
 // Utility function to check return from pi calls.
 // Throws if pi_result is not a PI_SUCCESS.
-// TODO: Absorb this utility in Trace Class
-template <typename Exception> inline void piCheckThrow(PiResult pi_result) {
+template <typename Exception = cl::sycl::runtime_error>
+inline void piCheckResult(PiResult pi_result) {
   CHECK_OCL_CODE_THROW(pi_result, Exception);
 }
 
-// Utility function to check if return from pi call is
-// PI_SUCCESS. If is it not, throw a cl::sycl::runtime_error.
-// TODO: Absorb this utility in Trace Class
-inline void piCheckResult(PiResult pi_result) {
-  piCheckThrow<cl::sycl::runtime_error>(pi_result);
-}
+#define PI_TRACE_ENABLED (std::getenv("SYCL_PI_TRACE") != nullptr)
 
-template <typename FnType> class Trace {
+template <typename FnType, size_t FnOffset> class Trace {
```

Comment:
Please add comments about what this class is doing, and I think it should be named "Call", and not "Trace" as this is its primary functionality. Also a good place to explain in comment what the template arguments are for.